pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -418,11 +427,22 @@ func (p *testPlanner) Plan() *TestPlan {
 		isLocal:        p.isLocal,
 	}
 
-	// Probabilistically enable some of of the mutators on the base test
-	// plan generated above.
+	failureInjections := make(map[string]struct{})
+	for _, m := range failureInjectionMutators {
+		failureInjections[m.Name()] = struct{}{}
+	}
+
+	// Probabilistically enable some of the mutators on the base test
+	// plan generated above. We disable any failure injections that
+	// would occur on clusters with less than three nodes as this
+	// can lead to uninteresting failures (e.g. a single node
+	// panic failure).
 	for _, mut := range planMutators {
 		if p.mutatorEnabled(mut) {
-			mutations := mut.Generate(p.prng, testPlan)
+			if _, found := failureInjections[mut.Name()]; found && !p.shouldWaitForReplication() {
```

Comment:
I'm confused why we are checking `!p.shouldWaitForReplication()` here. It's true that `p.shouldWaitForReplication()` always returns false for single node clusters, but you could have a three node cluster that disables `waitForReplication`